addons/website_blog/static/src/js/options.js

```diff
@@ -6,7 +6,7 @@ const {_t} = require('web.core');
 const options = require('web_editor.snippets.options');
 require('website.editor.snippets.options');
 
-if (!$('.website_blog').length) {
+if (!$('.o_iframe').contents().find('.website_blog').length) {
```

Comment:
I am not sure this actually is a good idea. I was not a fan of that before already but in this case I wonder if this is even right: are not those wysiwyg assets loaded only once in the backend? What if I am on a non blog page the first time, load the wysiwyg assets, leave edit mode, navigate to the blog page though the iframe and reenter edit mode... are the wysiwyg assets reloaded and this code reevaluated? Seems strange.

Reply:
@qsm-odoo Removed the "global" condition, and added a specific one on the override of `CoverProperties`'s `updateUI` override.
Also created task-3112890 to address the inconsistency between the selected option name and the patched option names induced by this method.

Reply:
@bso-odoo Nice but is what I was worried about correct?

- Iframe preview of homepage
- Edit _=> it is at this moment we load this options.js file, right?_
- Save/Discard
- Go to a blog post
- Edit _=> this options.js file is not reloaded, right?_

Reply:
@qsm-odoo Not exactly: the `options.js` is loaded even before entering edit mode - then loaded when entering edit - the loaded when leaving edit... on every page rendered while being logged in (on the `/@/...` route)
The condition as it was written in the previous iteration, when not in edit mode, did actually try to evaluate the content of the `iframe` BEFORE it was even loaded - which thus skipped the file. (`$('.o_iframe').contents()` did return an empty JQuery object)
The `options.js` file is part of `website.assets_wysiwyg` which is actively loaded by `createPublicRoot` when `loadWysiwyg` is set on the iframe. And in the `o_iframe` it is defined like this:
```xml
t-att-data-load-wysiwyg="this.websiteService.isRestrictedEditor ? 'true': 'false'"
```

The related commit (https://github.com/odoo/odoo/commit/a154ee7ad6fd3ebdd38943e1439badae11c3151d) says:
> The wysiwyg assets are now completely loaded in the frontend, after
creating the website root, when the frontend is displayed in an iframe.
It would be better if the frontend loads only what it needs of the
wysiwyg assets (some editor classes and the drop zones css), an
improvement would be to create an assets_wysiwyg_frontend or
assets_wysiwyg_minimal bundle.

Reply:
Meh, I guess we'll need to review things around that then but I'll see. Meanwhile, removing that global check is good IMO.